types/baseType.ts

```diff
@@ -0,0 +1,6 @@
+// Base Type for all Schema that enables soft-deletes for data recovery
+export interface IBaseType {
```

Comment:
oh wait one last thing - for consistency with not everything needing soft delete, should we separate this into two different bases?